### DISCUSSION TITLE
Exclude an exact match

### DISCUSSION THREAD

**Discussion by WhyNotHugo** [sharkdp/fd#1576] [Q&A] [answered]
I want to run `fzf` and exclude `Cargo.toml` without excluding other files that match this as a glob (e.g.: `subproject/Cargo.toml`).

Note that I'm actually constructing the command programmatically to "ignore the currently open file" in Vim ([full context](https://github.com/ibhagwan/fzf-lua/issues/396)), so I want a solution that works with arguments (e.g.: I can't add it to `./ignore`.

Is this currently viable?

**Comment by tavianator** [comment, marked as answer]
Could you use a pipeline?

```console
$ fd Cargo.toml
Cargo.toml
compiler/rustc_ast/Cargo.toml
compiler/rustc_ast_passes/Cargo.toml
...
$ fd Cargo.toml | grep -Fxv 'Cargo.toml'
compiler/rustc_ast/Cargo.toml
compiler/rustc_ast_passes/Cargo.toml
...
```

**Reply by WhyNotHugo** [reply]
I mostly wanted to avoid piping everything through a third process, but I guess `fd` can't do this by itself.

My invocation is done by neovim, so I used `pipeline` to avoid spawning a shell just for the pipe:

    pipeline fd cargo.toml '' grep -Fxv Cargo.toml

Thanks!

**Reply by tmccombs** [reply]
What is this pipeline command? I haven't seen that before.

**Reply by WhyNotHugo** [reply]
It's from execline. [pipeline](https://skarnet.org/software/execline/pipeline.html) runs two commands with a pipe between one's stdout and the other's stdin. It does so without spawning a whole shell (which would stay around until the children exit).

In this particular case, it helps me avoid having to wrap my entire command in `sh -c 'fd ... | grep ...'` (and deal with nested escaping of arguments).
